File: frontend/types/task.ts
import { Tag } from './tag';

export interface Task {
  id: number;
  title: string;
  description?: string;
  status: 'ToDo' | 'InProgress' | 'Done';
  priority: 'Low' | 'Medium' | 'High';
  dueDate?: string;
  createdAt: string;
  updatedAt: string;
  ownerId: number;
  owner?: {
    id: number;
    username: string;
    email: string;
  };
  tags?: Tag[];
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  priority?: 'Low' | 'Medium' | 'High';
  dueDate?: string;
}

export interface UpdateTaskRequest {
  title: string;
  description?: string;
  status?: 'ToDo' | 'InProgress' | 'Done';
  priority?: 'Low' | 'Medium' | 'High';
  dueDate?: string;
}

File: backend/TaskFlow.Api/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskFlow.Api.Models
{
    public enum TaskStatus
    {
        ToDo = 0,
        InProgress = 1,
        Done = 2,
    }

    public enum TaskPriority
    {
        Low = 0,
        Medium = 1,
        High = 2,
    }

    public class Task
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        public TaskStatus Status { get; set; } = TaskStatus.ToDo;

        public TaskPriority Priority { get; set; } = TaskPriority.Medium;

        public DateTime? DueDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Foreign Key to User (Owner)
        public int OwnerId { get; set; }

        // Navigation Property
        public User Owner { get; set; } = null!;

        // Navigation Property for many-to-many relationship with Tags
        public ICollection<TaskTag> TaskTags { get; set; } = new List<TaskTag>();
    }
}
